backend/app/health_trackers.go

```diff
@@ -0,0 +1,117 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"kubecloud/internal/constants"
+	"kubecloud/internal/logger"
+	"kubecloud/internal/notification"
+	"kubecloud/models"
+	"time"
+
+	proxy "github.com/threefoldtech/tfgrid-sdk-go/grid-proxy/pkg/client"
+	"github.com/xmonader/ewf"
+)
+
+func (h *Handler) TrackClusterHealth() {
+
+	interval := time.Duration(h.config.ClusterHealthCheckIntervalInHours) * time.Hour
+
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+	for range ticker.C {
+		logger.GetLogger().Info().Msg("Cluster health check test started")
+		clusters, err := h.db.ListAllClusters()
+		if err != nil {
+			logger.GetLogger().Error().Err(err)
+			continue
+		}
+
+		if len(clusters) == 0 {
+			logger.GetLogger().Info().Msg("No clusters to check health for")
+			continue
+		}
+
+		for _, cluster := range clusters {
+
+			wf, err := h.ewfEngine.NewWorkflow(constants.WorkflowTrackClusterHealth)
+			if err != nil {
+				logger.GetLogger().Error().
+					Err(err).
+					Msg("Failed to create health tracking workflow")
+				continue
+			}
+			cl, err := cluster.GetClusterResult()
+			if err != nil {
+				logger.GetLogger().Error().
+					Err(err).
+					Msg("Failed to get cluster result during health tracking")
+				continue
+			}
+			wf.State = ewf.State{
+				"cluster": cl,
+				"config": map[string]interface{}{
+					"user_id": cluster.UserID,
+				},
+			}
+
+			h.ewfEngine.RunAsync(context.Background(), wf)
+		}
+
+	}
+}
+
+func (h *Handler) TrackReservedNodeHealth(notificationService *notification.NotificationService, grid proxy.Client) {
+	interval := time.Duration(h.config.ReservedNodeHealthCheckIntervalInHours) * time.Minute
+
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		logger.GetLogger().Info().Msg("Reserved node health check started")
+
+		reservedNodes, err := h.db.ListAllReservedNodes()
+		if err != nil {
+			logger.GetLogger().Error().Err(err).Msg("Failed to get reserved nodes for health check")
+			continue
+		}
+
+		if len(reservedNodes) == 0 {
+			logger.GetLogger().Info().Msg("No reserved nodes to check health for")
+			continue
+		}
+
+		logger.GetLogger().Info().
+			Int("count", len(reservedNodes)).
+			Msg("Starting health check for reserved nodes")
+
+		for _, userNode := range reservedNodes {
```

Comment:
Shouldn't we use worker pool for this? and I also suggest we group them in the end by the user, instead of sending a separate report on each one